### PR TITLE
Release 0.27.1

### PR DESCRIPTION
**Pull request by bcardiff**
Preview at https://github.com/bcardiff/crystal/blob/changelog/0.27.1/CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,165 @@
...
+- Add SemanticVersion to API docs. ([#7003](https://github.com/crystal-lang/crystal/pull/7003), thanks @Blacksmoke16)
+- Add docs to discourage the use of `Bool#to_unsafe` other than for C bindings. ([#7320](https://github.com/crystal-lang/crystal/pull/7320), thanks @oprypin)
+- Refactor `#to_s` to be independent of the `name` method. ([#7295](https://github.com/crystal-lang/crystal/pull/7295), thanks @asterite)
+
+### Macros
+
+- Fixed docs of `ArrayLiteral#unshift`. ([#7127](https://github.com/crystal-lang/crystal/pull/7127), thanks @Blacksmoke16)
+- Fixed `Annotation#[]` to accept `String` and `Symbol` as keys. ([#7153](https://github.com/crystal-lang/crystal/pull/7153), thanks @MakeNowJust)
+- Fixed `NamedTupleLiteral#[]` to raise a compile error for invalid key type. ([#7158](https://github.com/crystal-lang/crystal/pull/7158), thanks @MakeNowJust)
+- Fixed `getter`/`property` macros to work properly with `Bool` types. ([#7313](https://github.com/crystal-lang/crystal/pull/7313), thanks @Sija) Add `read_file` macro method. ([#6967](https://github.com/crystal-lang/crystal/pull/6967), [#7094](https://github.com/crystal-lang/crystal/pull/7094), thanks @Sija, @woodruffw)
```
Comment (vladfaust) [inline CHANGELOG.md]:
```suggestion
- Fixed `getter`/`property` macros to work properly with `Bool` types. ([#7313](https://github.com/crystal-lang/crystal/pull/7313), thanks @Sija) 
- Add `read_file` macro method. ([#6967](https://github.com/crystal-lang/crystal/pull/6967), [#7094](https://github.com/crystal-lang/crystal/pull/7094), thanks @Sija, @woodruffw)
```

Comment (bcardiff) [reply CHANGELOG.md]: Thanks @vladfaust

**Comment by ysbaddaden** [comment]
Travis fails to run the Shards' test suite; it can't install the `crystal` package because it's not signed (or it doesn't know about the public signing key):
```
WARNING: The following packages cannot be authenticated!
  crystal
E: There were unauthenticated packages and -y was used without --allow-unauthenticated
```

**Comment by bcardiff** [comment]
@ysbaddaden signing of .deb packages have just been fixed.

**Comment by ysbaddaden** [comment]
Have overflow checks been enabled in 0.27.1 or on Travis specifically?

```
Unhandled exception: Overflow (OverflowError)
Failed to raise an exception: END_OF_STACK
```

**Comment by bcardiff** [comment]
@ysbaddaden in linux distro yes. The makefile in the compiler use `-D preview_overflow` and the distribution scripts use that makefile https://github.com/crystal-lang/distribution-scripts/blob/master/linux/Dockerfile#L98

**Comment by ysbaddaden** [comment]
Well, Crystal 0.27.1 fails to compile Shards, at least on Travis or it's the sat solver branch :(

**Pull request merged**